pwa/src/components/handlers/handlerForm.tsx

```diff
@@ -193,201 +193,243 @@ export default function HandlerForm({id, endpointId}) {
+                          {
+                            entities !== null && entities.length > 0 ? (
+                              <div className="form-group">
+                                {handler !== null &&
+                                handler.entity !== undefined &&
+                                handler.entity !== null ? (
+                                    <SelectInputComponent
+                                      options={entities}
+                                      data={handler.entity}
+                                      name={"entity"} id={"entityInput"}
+                                      nameOverride={"Entity"}
+                                      value={"/admin/entities/"}/>
+                                  )
+                                  : (
+                                    <SelectInputComponent
+                                      options={entities}
+                                      name={"entity"} id={"entityInput"}
+                                      nameOverride={"Entity"}
+                                      value={"/admin/entities/"}/>
+                                  )}
+                              </div>
+                            ) : (
+                              <SelectInputComponent
+                                options={[]}
+                                name={"entity"} id={"entityInput"}
+                                nameOverride={"Entity"}
+                                value={"/admin/entities/"}/>
+                            )
+                          }
+                        </div>
+                        <div className="col-6">
+                          {handler !== null && handler.type !== null ? (
+                            <SelectInputComponent
+                              options={[
+                                {name: "twig", value: "twig"},
+                                {name: "markdown", value: "markdown"},
+                                {name: "restructuredText", value: "restructuredText"},
+                              ]}
+                              name={"templateType"}
+                              id={"templateTypeInput"}
+                              nameOverride={"Template Type"}
+                              data={handler.templateType}
+                            />
+                          ) : (
+                            <SelectInputComponent
+                              options={[
+                                {name: "twig", value: "twig"},
+                                {name: "markdown", value: "markdown"},
+                                {name: "restructuredText", value: "restructuredText"},
+                              ]}
+                              name={"type"}
+                              id={"templateTypeInput"}
+                              nameOverride={"Template Type"}
+                            />
+                          )}
```

Comment:
This entire block can be simplified I think.

Basically, we're checking if we have data for the select element and based on that we render another select element. However, we could be rendering the same select element in both cases, and only populate it with data when we have it.

So, at the top of this block we're doing the following check: `{handler !== null && handler.type !== null ? (`. This entire check can be removed (and therefore we only need the code within the `:` and we can remove the block within the `?`.

Finally, because we've removed the check. We do not know if we have the data to populate the field. Therefore, we have to rewrite this line: `data={handler.templateType}` to `data={handler?.templateType}` the `?` after the variable `handler` means `if (handler) { handler.data }`.

So above means: if there's no handler, we just leave data empty. If there is a handler, we fill the data with handler.templateType.